test/linalg/test_linalg.fypp

```diff
@@ -552,6 +562,336 @@ contains
             "trace(h) == sum(c(0:nd:2)) failed.")
 
     end subroutine test_trace_int64
+    
+    subroutine test_kronecker_product_rsp(error) 
```

Comment:
I suggest to move all these tests in a fypp loop. This would avoid all these copy-pastes and possible errors/typos.

Reply:
I see. I could not find the fypp loop for the other tests to be generated. The reason I assumed the other tests weren't in a loop is because there seemed to be special behavior for checking if there is quad_precision. I did, however, generate the subroutines using an fypp loop before adding the if-statements for checking for quad precision. 
Do you have a suggestion for me for this? If I loop over all supported types normally, it won't generate quad_precision when someone makes stdlib?

Reply:
Just generate a loop with the REAL_KINDS_TYPES. If QP is not supported, it it not added to the latter variable. See [common.fypp](https://github.com/fortran-lang/stdlib/blob/master/src/common.fypp) for the details and also [test_mean.fypp](https://github.com/fortran-lang/stdlib/blob/master/test/stats/test_mean.fypp) for an example.

Reply:
Got it. I condensed the test into a single fypp loop and also made the expected array a parameter. Rebuilt and reran the tests, all passed. Checked the tests .f90 file, all tests seem to have generated fine.